### PR TITLE
#160604147 - Setup Travis

### PR DESCRIPTION
**Pull request by andrewinsoul**
### What does this PR do?
This PR setups travis and add travis, coveralls and code-climate badges to README

### Description of Task to be completed
- Install nyc and modify test script
- add coverage and run-coverage script
- populate README and add badges to it
- setup Travis ci for automated building

### How should this be manually testes?
Go to README and see a travis pass badge in it

### What are the relevant Pivotal Tracker Stories?
#160604147

### CODE REVIEW

File: server/middlewares/helperFunction.js [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
+    order => order.id === Number(req.params.id)
+  );
   const error = 'order not found';
-  if (index === -1) return res.status(404).send({ status: 'error', error });
+  if (index === -1) {
+    return res.status(404).send({ status: 'error', error });
+  }
   return next();
 }
 export default Helper;
```
Comment (dogoyaro) [inline server/middlewares/helperFunction.js]: The name of this export has to be the same as the name of the file i.e `helperFunction.js`. I also don't think helperFunction is a descriptive enough name. What does this helper function do?

File: server/test/test.js [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+            address: 'address',
+          })
+          .end((err, res) => {
+            expect(res).to.have.status(400);
+            expect(res.body.status).to.eql('error');
+            expect(res.body.error).to.eql('order field cannot be blank');
+            done();
+          });
+      });
+      it('should return 400 with error message', (done) => {
```
Comment (dogoyaro) [inline server/test/test.js]: When should this happen. Test descriptions should read `it should do this when that has happened`. Please make sure this applies to all of them

File: server/test/test.js [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+            expect(res.body.status).to.eql('error');
+            expect(res.body.error).to.eql(
+              'only strings are allowed for the order field'
+            );
+            done();
+          });
+      });
+    });
+
+    describe('tests for invalid inputs in updating the status endpoint', () => {
```
Comment (dogoyaro) [inline server/test/test.js:178]: Please use the proper test description. `it should ... when ... `

File: server/middlewares/validateFunction.js [hunk 4]
```diff
@@ -61,8 +139,15 @@ class Validator {
    * @returns {object} - status code and error message
    */
   getOrder(req, res, next) {
-    if (isNaN(req.params.id)) return res.status(400).send({ status: 'error', error: 'params shoudld be a number' });
-    if (!(parseInt(req.params.id, 10) === Number(req.params.id))) return res.status(400).send({ status: 'error', error: '' });
+    if (isNaN(req.params.id)) {
+      return res.status(400).send({
+        status: 'error',
+        error: 'params shoudld be a number'
```
Comment (dogoyaro) [inline server/middlewares/validateFunction.js]: Typo here. Also, be more specific with the error message. What `params` are you referring to?

**Review by dogoyaro — CHANGES_REQUESTED** [review]

**Comment by dogoyaro** [comment]
There are multiple commits in this PR. Please fix before merging into develop.

File: server/middlewares/helperFunction.js (hunk 1 above)
Comment (andrewinsoul) [reply server/middlewares/helperFunction.js]: The function helps me prevent code duplication and it basically checks if a resource exists when the GET and PUT endpoints are accessed

File: server/middlewares/validateFunction.js [hunk 5]
```diff
@@ -61,8 +139,15 @@ class Validator {
    * @returns {object} - status code and error message
    */
   getOrder(req, res, next) {
-    if (isNaN(req.params.id)) return res.status(400).send({ status: 'error', error: 'params shoudld be a number' });
-    if (!(parseInt(req.params.id, 10) === Number(req.params.id))) return res.status(400).send({ status: 'error', error: '' });
+    if (isNaN(req.params.id)) {
+      return res.status(400).send({
+        status: 'error',
+        error: 'id params shoudld be a number'
```
Comment (dogoyaro) [inline server/middlewares/validateFunction.js]: I pointed out this typo before. It doesn't seem like you fixed it `shoudld` should be `should`.

**Review by dogoyaro — CHANGES_REQUESTED** [review]

**Pull request merged**